src/main/java/it/gov/pagopa/pu/fileshare/connector/processexecutions/client/IngestionFlowFileClient.java

```diff
@@ -30,4 +33,21 @@ public String createIngestionFlowFile(IngestionFlowFileRequestDTO ingestionFlowF
       throw e;
     }
   }
+
+  public IngestionFlowFile getIngestionFlowFile(Long ingestionFlowFileId, String accessToken) {
+    try {
+      log.debug("Fetching ingestion flow file with ID [{}]", ingestionFlowFileId);
+      return processExecutionsApisHolder.getIngestionFlowFileEntityControllerApi(accessToken).crudGetIngestionflowfile(String.valueOf(ingestionFlowFileId));
+    } catch (HttpClientErrorException e) {
+      log.error("Error fetching ingestion flow file with ID [{}]", ingestionFlowFileId, e);
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
+        throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Ingestion Flow File not found");
```

Comment:
inside a connector it's better to return null instead!
if null should return to the Controller an exception to be transformed into 404 it's something that the service should perform throwing a custom exception, then converted through the RestControllerAdvice into the HTTP status required